### DISCUSSION TITLE
Keeping track of how long someone stays on a page

### DISCUSSION THREAD

**Discussion by jwendel** [plausible/analytics#1743] [Self-Hosted Support] [answered]
I just started using Plausible for a small site I have, and I have a question (this may just be ignorance on configuring analytics).  My site is a single tool that just reports status of some other service and refreshes the data on the page every minute (just with some simple javascript and an XHR call to the server to get the data).  From what I can tell, people will keep the page open for a while, and when they are all done, will close the tab.  They will never navigate to another page.

I saw the "time on page" discussion https://github.com/plausible/analytics/discussions/111, which makes it look like the `sendBeacon` call may do what I want, but I may just wait for you to implement this, as it sounds like it may be flaky.

Today I have this XHR call I'm doing every minute, so the question is, which of these 2 is better?

1) Use the [Events API](https://plausible.io/docs/events-api) to also ping Plausible when the XHR status update call comes to the server.
2) When I do the XHR call to get the data, use the [Custom Event](https://plausible.io/docs/custom-event-goals) to record the user still being there

My guess is (1) may cause me to record calls where a bot or something similar is calling the endpoint (as you [talked about here](https://plausible.io/blog/server-log-analysis)).  But (2) has the downside of an extra call for the client.

Do I have a fair reading of my options and pro/cons?

**Comment by ukutaht** [comment, marked as answer]
Hey @jwendel welcome!

Pinging the server will extend the _visit duration_ metric but it will not do anything for _time on page_. Since time on page is calculated as the difference between two pageviews in the database, custom events do not affect the metric.

**Reply by jwendel** [reply]
Good to know!  I'm guessing triggering a pageview for the same URL every minute would also break the "time on page" stat?

Aside: I used the typescript module for Plausible and incorporated pageview + event into my existing JS, and it worked wonderfully.

**Reply by ukutaht** [reply]
> Good to know! I'm guessing triggering a pageview for the same URL every minute would also break the "time on page" stat?

Yeah it won't improve the time on page unfortunately.

Great to hear that the integration was smooth!
